app/src/main/java/com/asfoundation/wallet/billing/address/BillingAddressPresenter.kt

```diff
@@ -1,23 +1,36 @@
 package com.asfoundation.wallet.billing.address
 
+import com.asfoundation.wallet.billing.analytics.BillingAnalytics
 import io.reactivex.Scheduler
 import io.reactivex.disposables.CompositeDisposable
 
 class BillingAddressPresenter(
     private val view: BillingAddressView,
+    private val data: BillingAddressData,
+    private val navigator: BillingAddressNavigator,
+    private val billingAnalytics: BillingAnalytics,
     private val disposables: CompositeDisposable,
     private val viewScheduler: Scheduler) {
 
   fun present() {
+    initializeView()
     handleSubmitClicks()
     handleBackClicks()
   }
 
+  private fun initializeView() {
+    view.initializeView(data.bonus, data.isDonation, data.domain, data.skuDescription,
+        data.appcAmount, data.fiatAmount, data.fiatCurrency, data.isStored, data.shouldStoreCard)
+  }
+
   private fun handleSubmitClicks() {
     disposables.add(
         view.submitClicks()
             .subscribeOn(viewScheduler)
-            .doOnNext { view.finishSuccess(it) }
+            .doOnNext { billingAddressModel ->
+              sendActionEventAnalytics(if (data.isDonation) "donate" else "buy")
```

Comment:
In general, I agree with not having hardcoded strings. In this case, not only does it already happen in other cases (look at any other presenter like "CodeValidationDialogPresenter"), but it is also very specific to the action itself (the action parameter in the event should be the same as the button text in english). So while I agree in principle, I think in theses cases it is more readable to be hardcoded (and since there's no reusing the same string, there's no benefit either)